Fix PG behaviour when specifying timestamps in the past

leplatrem opened the pull request:
I rewrote cleanly 4 four tests for storage backends:
- create with timestamp specified in future and past
- update with timestamp specified in future and past

I fixed the migrations of PR #604 

And I updated the migration of PR #649 to fix the behaviour of PG for collection timestamps when specified record timestamp were in the past.

Natim commented:
I don't understand why you had to modify the 008_to_009 migration file. However it looks good to me.

leplatrem commented:
> I don't understand why you had to modify the 008_to_009 migration file

In PR #604, we updated the SQL schema, but didn't bump its version. As a consequence, the migration 008-009 was never executed.

In PR #649, I replaced the orphan migration file 008-009 with a new one about collection timestamps (my fault).

This PR restores the migration of #604 on 008-009 and puts the one of #649 into a new one 009-010.

Natim commented:
Ok makes sense :+1:

leplatrem commented on cliquet_docs/api/resource.rst:
```diff
@@ -424,6 +424,18 @@ A ``details`` attribute in the response provides the offending record and
+``last_modified`` attribute.
+
+The specified value will be ignored if it is less than the current collection
+timestamp.
```
Incorrect statement.

leplatrem commented on cliquet_docs/api/resource.rst:
```diff
@@ -567,17 +583,23 @@ The consumer might decide to ignore it.
+querystring with ``?last_modified=<value>``.
+
+The specified value will be ignored if it is less than the current collection
+timestamp.
```
Incorrect statement.

leplatrem commented on cliquet_docs/api/resource.rst:
```diff
@@ -651,6 +674,20 @@ changed meanwhile, a ``412 Precondition failed`` error is returned.
+
+The specified value will be ignored if:
+
+* it is less than the current collection timestamp;
```
Incorrect statement.

leplatrem commented on cliquet_docs/api/resource.rst:
```diff
@@ -758,6 +799,21 @@ If changing a record field violates a field unicity constraint, a
+
+The specified value will be ignored if:
+
+* it is less than the current collection timestamp;
```
Incorrect statement.

leplatrem commented on cliquet_docs/api/timestamps.rst:
```diff
@@ -68,3 +68,20 @@ The client can then choose to:
+
+It will be ignored if:
+
+* it is less than the current collection timestamp;
```
Incorrect statement.

Natim commented:
r+
